includes/class-freemius.php

```diff
@@ -629,6 +636,254 @@ private function _version_updates_handler() {
             }
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.0
+         *
+         * @param string $sdk_prev_version
+         * @param string $sdk_version
+         */
+        function _handle_gdpr_admin_notice( $sdk_prev_version, $sdk_version ) {
+            /**
+             * @since 1.1.7.3 Fixed unwanted connectivity test cleanup.
+             */
+            if ( empty( $sdk_prev_version ) ) {
+                return;
+            }
+
+            if ( version_compare( $sdk_prev_version, '2.1.0', '<' ) &&
+                version_compare( $sdk_version, '2.1.0', '>=' )
+            ) {
+                add_action( 'init', array( &$this, '_maybe_show_gdpr_admin_notice' ) );
+            }
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.0
+         */
+        function _maybe_show_gdpr_admin_notice() {

```

Comment:
@fajardoleo since this method is executed per product, it involves API calls which can take time. So we can get into a theoretical situation where an admin opens the admin page several times in parallel right after an update of the SDK, which can run this logic several times. Unfortunately, PHP doesn't have locks/mutexes. BUT, we can significantly reduce the chance of that happening by simulating a lock using the storage.

I'm basically suggesting to add the following logic right after `if ( ! $this->is_user_admin() ) {` and once we have `$current_wp_user`:
```php
    $user_id = $current_wp_user->ID;

    if ( get_transient( "locked_{$user_id}" ) ){
         return;
    }

    set_transient( "locked_{$user_id}", true,  20); // 20-sec lock.
```

That way we make sure that each admin will only execute this method once in 20 sec, max.